src/Sylius/Component/Core/Model/User.php

```diff
@@ -81,4 +87,62 @@ public function getRoles()
 
         return $roles;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCustomer()
+    {
+        return $this->customer;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setCustomer(BaseCustomerInterface $customer = null)
+    {
+        if ($this->customer !== $customer) {
+            $this->customer = $customer;
+            $this->assignUser($customer);
```

Comment:
Might be only my personal feeling, but I'd inject the innards of this method in here, because User::assignUser which takes a Customer instance and is used in a setCustomer method looks quite freaky